pilot/pkg/bootstrap/server.go

```diff
@@ -346,7 +347,7 @@ type startFunc func(stop <-chan struct{}) error
 // initMonitor initializes the configuration for the pilot monitoring server.
 func (s *Server) initMonitor(args *PilotArgs) error { //nolint: unparam
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		monitor, addr, err := startMonitor(args.DiscoveryOptions.MonitoringAddr, s.mux)
+		monitor, addr, err := istiod.StartMonitor(args.DiscoveryOptions.MonitoringAddr, s.mux)
```

Comment:
pilot should not depend on istiod

Reply:
So can we use these repeated codes as public libraries? In this way, the repetition of the whole file will bring difficulties to the  code reading later.